util/minio_client.py

```
@@ -0,0 +1,36 @@
+import logging
+from datetime import timedelta
+
+from django.conf import settings
+from django.utils.functional import cached_property
+from minio import Minio
+
+logger = logging.getLogger(__name__)
+
+
+class MinIOClient:
+    @cached_property
+    def client(self):
+        return Minio(
+            settings.MINIO_DATA["endpoint"],
+            access_key=settings.MINIO_DATA["accessKey"],
+            secret_key=settings.MINIO_DATA["secretKey"],
+            secure=True,
+        )
+
+    def get_presigned_url(self, mesh_id, expires=timedelta(hours=1)):
+        if not mesh_id:
+            return None
```

maybe you need to raise an error here too, rather than None. Not sure with the dataflow you need though